src/server/src/main/java/io/cassandrareaper/resources/ClusterResource.java

```diff
@@ -322,88 +264,59 @@ public Response modifyClusterSeed(
    * @param cluster the Cluster object we intend to update
    * @param seedHosts a list of hosts to connect to in the cluster
    * @return the updated cluster object with a refreshed seed list
-   * @throws ReaperException Any runtime exception that could be triggered
+   * @throws ReaperException failure to jmx connect/call to cluster
    */
   private Cluster updateClusterSeeds(Cluster cluster, String seedHosts) throws ReaperException {
     Set<String> newSeeds = parseSeedHosts(seedHosts);
-
     try {
-      JmxProxy jmxProxy =
-          context.jmxConnectionFactory.connectAny(
-              Optional.absent(),
-              newSeeds
-                  .stream()
-                  .map(
-                      host ->
-                          Node.builder()
-                              .withClusterName(cluster.getName())
-                              .withHostname(parseSeedHost(host))
-                              .build())
-                  .collect(Collectors.toList()),
-              context.config.getJmxConnectionTimeoutInSeconds());
-
+      JmxProxy jmxProxy = connectAny(newSeeds,cluster.getName());
       Optional<List<String>> liveNodes = Optional.of(jmxProxy.getLiveNodes());
       newSeeds = liveNodes.get().stream().collect(Collectors.toSet());
-
-      if (cluster.getSeedHosts().equals(newSeeds)) {
-        // No change in the node list compared to storage
-        return cluster;
+      if (!cluster.getSeedHosts().equals(newSeeds)) {
+        cluster = new Cluster(cluster.getName(), cluster.getPartitioner(), newSeeds);
+        context.storage.updateCluster(cluster);
       }
-
-      Cluster newCluster = new Cluster(cluster.getName(), cluster.getPartitioner(), newSeeds);
-      context.storage.updateCluster(newCluster);
-
-      return newCluster;
-
+      return cluster;
     } catch (ReaperException e) {
-      throw new ReaperException("failed to create cluster with new seed hosts", e);
+      throw new ReaperException(
+          String.format("failed to update cluster %s with new seed hosts %s", cluster.getName(), seedHosts),
+          e);
     }
   }
 
   /**
    * Delete a Cluster object with given name.
    *
    * <p>
-   * Cluster can be only deleted when it hasn't any RepairRun or RepairSchedule instances under it, i.e. you mus
+   * Cluster can be only deleted when it hasn't any RepairRun or RepairSchedule instances under it, i.e. you must
    * delete all repair runs and schedules first.
    *
    * @param clusterName The name of the Cluster instance you are about to delete.
-   * @return The deleted RepairRun instance, with state overwritten to string "DELETED".
    */
   @DELETE
   @Path("/{cluster_name}")
   public Response deleteCluster(
-      @PathParam("cluster_name") String clusterName)
-      throws ReaperException {
+      @PathParam("cluster_name") String clusterName) {
 
-    LOG.info("delete cluster called with clusterName: {}", clusterName);
+    LOG.info("delete cluster {}", clusterName);
```

Comment:
Nit pick: Perhaps change this to say something like:

> "Deleting cluster {}"

Reply:
the verb isn't appropriate as that isn't yet confirmed.